**Az első 5 feladat ismétlés (többtáblás lekérdezések) a tanulmányi adatbázisból**

1. Készítsünk listát arról, hogy melyik óra melyik teremben van!
    
    1. Jelenítsük meg az óra azonosítóját és a terem nevét!
    2. A listát szűrjük a 3 karakter hosszú, \[100;199\] zárt intervallumba eső termekre! Feltesszük, hogy a 3 karakter hosszú terem nevek csak számjegyeket tartalmaznak.

In [1]:
SELECT o.ora_id, t.terem
FROM Orak o JOIN Termek t ON o.terem=t.tkód
WHERE LEN(t.terem)=3 AND t.terem BETWEEN 100 AND 199

(30 rows affected)

Total execution time: 00:00:00.048

ora_id terem 159 117 160 118 161 118 162 119 163 120 164 119 165 119 166 119 167 120 170 120 172 119 173 119 175 120 176 118 177 119 179 120 180 119 181 118 185 120 192 119 195 116 197 116 198 119 204 120 212 120 218 118 220 119 222 116 226 117 227 116

2. <span style="font-weight: 600; color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px;">Hány óra van összesen az egyes napokon?</span>

1. Jelenítsük meg a napok nevét és az órák számát az adott napon!
2. Az oszlopokat nevezzük el értelemszerűen!
3. A listát rendezzük óraszám szerint növekvő sorrendbe!

In [17]:
SELECT n.nap,COUNT(*) AS "Órák száma"
FROM Orak o JOIN Napok n ON o.nap=n.nap_id
GROUP BY n.nap
ORDER BY "Órák száma" ASC;


(5 rows affected)

Total execution time: 00:00:00.011

nap Órák száma Péntek 6 Szerda 6 Kedd 13 Csütörtök 19 Hétfő 23

3. <span style="font-weight: 600; color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px;">Készítsünk listát arról, hogy melyik oktatónak hány órája van az Órák táblában!</span>

1. A lista jelenítse meg az oktatók nevét, és az órák számát!
2. Az oszlopokat nevezzük el értelemszerűen!
3. A listából hagyjuk ki az óraadókat!

In [54]:
SELECT ok.nev,COUNT(*) AS 'Óraszám'
FROM Orak o JOIN Oktatok ok ON o.tanar=ok.oktato_id
            JOIN statusok s ON ok.[status]=s.Skod
WHERE NOT s.[Status] = 'Óraadó'
GROUP BY ok.nev

(18 rows affected)

Total execution time: 00:00:00.009

nev Óraszám Aranyi Mária 1 Balkán András 3 Jogrerős Pál 2 Juhász András 2 Kiss János 4 Kovács András 4 Kovács László 4 Ludas Mátyás 2 Nógrádi Kata 1 Róbert Gida 1 Szabó Andrera 2 Szénási ászló 2 Szentágotai Istvás 8 Szép llonka 4 Szerb László 3 Ujfalussi István 3 Varga Kata 4 Vitéz János 4

4. <span style="color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px;">Van-e olyan tantárgy, amelyből nincs óra?</span>

1. Jelentsük meg a tantárgy nevét és a tantárgyfelelős nevét!

In [59]:
SELECT t.Tantárgynév,ok.nev
FROM Tantárgyak t LEFT JOIN Orak o ON t.Tkód = o.targy
                        JOIN Oktatok ok ON t.Tantárgyfelelős=ok.oktato_id
WHERE o.ora_id is NULL

(1 row affected)

Total execution time: 00:00:00.008

Tantárgynév nev Folyami gáttan Varga Kata

5. <span style="background-color: rgb(255, 255, 255); color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px;">Hány óra van összesen a 120-as teremben?</span>

**A 6-10 feladatokat halmazműveletek segítségével oldjuk meg, szintén a tanulmányi adatbázison**

6. <span style="background-color:rgb(255, 255, 255);color:rgb(36, 41, 47);font-family:-apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;;font-size:16px;font-weight:600;">Jelenítsük meg azon tanárok azonosítóját és nevét, akik hétfői vagy keddi napokon tanítanak!</span>

7. <span style="background-color: rgb(255, 255, 255); color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-weight: 600;">Melyek azok a tantárgyak, amelyek esetén van óra a 8.00-kor kezdődő sávban, de nincs a 12.30-kor kezdődő sávban?</span>

8. <span style="background-color: rgb(255, 255, 255); color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-weight: 600;">Listázzuk azokat a termeket, amelyekben oktat Kovács László és van olyan tanóra a teremben, ahol a tantárgy nevében nem szerepel az 'ürge' szó!</span>

9. <span style="font-weight: 600; color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px;">Készítsünk listát arról, hogy melyik tanárnak hány órája van a (munka) hét elején (hétfőn), hét közben (kedd, szerda és csütörtök), illetve a hét végén (péntek)!</span>

1. Jelenítsük meg a tanár azonosítóját, nevét, az időszakot (hét eleje - hét közben - hét vége), valamint az óraszámot!
2. A három megjelölt időszakot külön kérdezzük le, majd fűzzük össze a lekérdezések eredményeit!
3. Rendezzük a listát az oktató neve, azon belül időszak szerint!

10. <span style="background-color: rgb(255, 255, 255); color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-weight: 600;">Melyik napokon nincs órája Pelikán Józsefnek?</span>

In [62]:
SELECT nap
FROM Napok
EXCEPT
SELECT n.nap
FROM Orak o JOIN Oktatok ok ON o.tanar =ok.oktato_id
            JOIN Napok n ON o.nap=n.nap_id
WHERE ok.nev ='Pelikán József'

(3 rows affected)

Total execution time: 00:00:00.015

nap Csütörtök Péntek Szerda

**A 11-15 feladatok gyakorlásra szolgálnak a közelgő zh-ra :-)**

11. _Készítsünk listát az oktatók adatairól!_

1. Az oktatók titulusa és neve összefűzve jelenjen meg, közöttük szóközzel!
2. Ha valakinek nincs titulusa, akkor csak a neve jelenjen meg, szóköz nélkül!
3. A listát rendezzük a beosztás kódja, azon belül a státusz kódja szerint növekvő sorrendbe!
4. Az oszlopokat nevezzük el értelemszerűen!

In [10]:
SELECT IIF(titulus IS NUll, nev,titulus+' '+nev) AS 'Név',beosztás,status
FROM Oktatok
ORDER BY beosztás,status


(25 rows affected)

Total execution time: 00:00:00.020

Név beosztás status Nógrádi Kata 1 1 Rácz András 1 1 Kovács András 1 1 Ludas Mátyás 1 1 Aranyi Mária 1 1 Szabó Andrera 1 1 Kiss István 1 2 Mészáros Kriszta 1 2 Róbert Gida 2 1 Vitéz János 2 1 Kiss János 2 1 Pelikán József 2 3 Farkasos Piroska 2 3 Dr. Szép llonka 3 1 Szénási ászló 3 1 Dr. Szentágotai Istvás 3 2 Szerb László 4 1 Dr. Józsa Gábor 4 1 Dr. Varga Kata 4 1 Dr. Balkán András 4 2 Dr. Kovacs Istvan 5 1 Dr. Juhász András 6 1 Prof. Jogrerős Pál 6 1 Dr. Kovács László 6 2 Prof, tanszékvezető Ujfalussi István 6 2

<span style="color: #09885a;">12</span>. Készítsünk listát, amely megmutatja, hogy összesen hány tantárgy van kreditenkénti csoportosításban!

  

A listából hagyjuk ki azokat a tételeket, ahol ez a darabszám <span style="color: #09885a;">1</span>!

A Folyami gáttan tantárgyat ne vegyük figyelembe!

A listát rendezzük darabszám szerint csökkenő sorrendbe!

Az oszlopokat nevezzük el értelemszerűen!

In [31]:
SELECT COUNT(Tantárgynév) AS 'Tantárgyszám', Kredit AS 'Kreditszám'
FROM Tantárgyak
WHERE Tantárgynév <> 'Folyami gáttan'
GROUP BY Kredit
HAVING COUNT(Tantárgynév) >1
ORDER BY 'Tantárgyszám' DESC


(4 rows affected)

Total execution time: 00:00:00.024

Tantárgyszám Kreditszám 3 2 3 3 2 4 2 1

13. _Készítsünk listát az oktatók nevéről, a beosztásuk alapján elvárt és a tényleges óraszámukról!_

1. Az oszlopok neve legyen 'Oktató neve', 'Elvárt óraszám' és 'Tényleges óraszám'
2. Szűrjük a listát azokra az oktatókra, ahol az elvárt és a tényleges óraszám azonos

In [57]:
SELECT o.nev,COUNT(b.elvart) AS 'elvárt', COUNT(o.nev) AS 'valós'
FROM Oktatok o JOIN Orak ora ON o.oktato_id=ora.tanar
JOIN Beosztasok b ON o.beosztás=b.beosztas_id
GROUP BY o.nev , b.elvart
HAVING COUNT(b.elvart)=COUNT(o.nev)

                

(20 rows affected)

Total execution time: 00:00:00.027

nev elvárt valós Aranyi Mária 1 1 Balkán András 3 3 Farkasos Piroska 7 7 Jogrerős Pál 2 2 Juhász András 2 2 Kiss János 4 4 Kovács András 4 4 Kovács László 4 4 Ludas Mátyás 2 2 Nógrádi Kata 1 1 Pelikán József 6 6 Róbert Gida 1 1 Szabó Andrera 2 2 Szénási ászló 2 2 Szentágotai Istvás 8 8 Szép llonka 4 4 Szerb László 3 3 Ujfalussi István 3 3 Varga Kata 4 4 Vitéz János 4 4

14. _Készítsünk listát, amely megmutatja, hogy mely tantárgyból nincs óra egyáltalán!_

1. Csak a tantárgy neve jelenjen meg!

In [66]:
SELECT t.Tantárgynév
FROM Tantárgyak t LEFT JOIN Orak o ON t.Tkód=o.targy
WHERE o.ora_id is NULL

(1 row affected)

Total execution time: 00:00:00.114

Tantárgynév Folyami gáttan

15. _Hány órájuk van az óraadóknak napi bontásban?_

1. Csak a napok nevét és az óraszámot jelenítsük meg!
2. Rendezzük a listát óraszám szerint növekvő sorrendbe!

In [71]:
SELECT n.nap,COUNT(*)
FROM Orak o JOIN Oktatok okt on o.tanar=okt.oktato_id JOIN Napok n on n.nap_id=o.nap JOIN statusok s on okt.[status]=s.Skod
WHERE s.[Status]='Óraadó'
GROUP BY n.nap
ORDER BY 2

(5 rows affected)

Total execution time: 00:00:00.033

nap (No column name) Péntek 1 Szerda 1 Csütörtök 3 Kedd 3 Hétfő 5